Fixing build with as-needed, respecting LDFLAGS
http://bugs.gentoo.org/show_bug.cgi?id=320795
Also fix underlinking, adding missing -lX11
http://bugs.gentoo.org/show_bug.cgi?id=369889

--- Makefile
+++ Makefile
@@ -7,14 +7,14 @@
 PIXMAP_DIR	:= $(PREFIX)/share/pixmaps
 MANPAGE_DIR	:= $(PREFIX)/share/man/man1
 
-CFLAGS 	+= $(shell pkg-config audacious --cflags) $(shell pkg-config dbus-1 --cflags)
-LIBS 	:= $(shell pkg-config audacious --libs)   $(shell pkg-config dbus-1 --libs)
+CFLAGS 	+= $(shell pkg-config audclient --cflags) $(shell pkg-config dbus-1 --cflags)
+LIBS 	:= $(shell pkg-config audclient --libs)   $(shell pkg-config dbus-1 --libs) -lX11
 
 OBJS 	= wmauda.o
 HEADERS = config.h dock-master.xpm
 
 wmauda:	$(OBJS) $(HEADERS)
-	$(CC) -o wmauda $(OBJS) $(CFLAGS) $(LIBS)
+	$(CC) $(LDFLAGS) -o wmauda $(OBJS) $(CFLAGS) $(LIBS)
 
 all:	wmauda
 
